TeamCode/src/main/java/org/firstinspires/ftc/teamcode/rework/opmodes/VisionTest.java

```diff
@@ -0,0 +1,86 @@
+package org.firstinspires.ftc.teamcode.rework.opmodes;
+
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
+
+import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
+import org.firstinspires.ftc.teamcode.rework.vision.RingStackLocator;
+import org.opencv.core.Mat;
+import org.openftc.easyopencv.OpenCvCameraFactory;
+import org.openftc.easyopencv.OpenCvCameraRotation;
+import org.openftc.easyopencv.OpenCvPipeline;
+import org.openftc.easyopencv.OpenCvWebcam;
+
+@TeleOp
+public class VisionTest extends LinearOpMode {
+    OpenCvWebcam webcam;
+
+    @Override
+    public void runOpMode()
+    {
+        VisionTestPipeline pipeline = new VisionTestPipeline();
+        webcam = OpenCvCameraFactory.getInstance().createWebcam(hardwareMap.get(WebcamName.class, "Webcam 1"));
+
+        webcam.setPipeline(pipeline);
+
+        webcam.setViewportRenderingPolicy(OpenCvWebcam.ViewportRenderingPolicy.OPTIMIZE_VIEW);
+        webcam.openCameraDeviceAsync(() -> {
+            webcam.startStreaming(320, 240, OpenCvCameraRotation.UPRIGHT);
+        });
+
+        telemetry.addLine("Waiting for start");
+        telemetry.update();
+
+        waitForStart();
+
+        while (opModeIsActive())
+        {
+            telemetry.addData("Frame Count", webcam.getFrameCount());
+            telemetry.addData("FPS", String.format("%.2f", webcam.getFps()));
+            telemetry.addData("Total frame time ms", webcam.getTotalFrameTimeMs());
+            telemetry.addData("Pipeline time ms", webcam.getPipelineTimeMs());
+            telemetry.addData("Overhead time ms", webcam.getOverheadTimeMs());
+            telemetry.addData("Theoretical max FPS", webcam.getCurrentPipelineMaxFps());
+            telemetry.addData("Calculated number of rings in stack", pipeline.numRings);
+
+            telemetry.update();
+
+            if(gamepad1.a)
+            {
+                webcam.stopStreaming();
+                //webcam.closeCameraDevice();
+            }
+
+            sleep(100);
+        }
+    }
+
+    class VisionTestPipeline extends OpenCvPipeline
+    {
+        public RingStackLocator.TARGET_ZONE numRings = RingStackLocator.TARGET_ZONE.TARGET_ZONE_A;
```

Comment:
Misleading variable name? Also why is this hard coded

Reply:
default value so telemetry doesn't throw a nullpointerexception

Reply:
Perhaps add an Unknown state?